eta/core/datasets.py

```diff
@@ -1582,111 +1625,118 @@ def _to_helper_list_video_objects(self, records):
+        return int(np.quantile(np.sort(counts), self.target_quantile))
 
-        # sort ascending by count
-        counts_list = counts.most_common()[::-1]
-
-        count_vals = [v for k, v in counts_list]
-        target_count = int(np.quantile(count_vals, self.target_quantile))
-
-        target_remove = {k: v - target_count for k, v in counts_list}
-        # @TODO leave the negatives in? (comment this line out)
-        # target_remove = {k: max(v, 0) for k, v in target_remove.items()}
-
-        return target_remove
-
-    def _get_keep_idxs(self, target_remove, helper_list):
+    def _get_keep_idxs(self, A, counts, target_count):
         '''Algorithm fun! This function chooses the set of records to keep (and
         remove).
 
         There's still plenty of potential for testing and improvement here.
 
+
+        This problem can be posed as:
+            minimize:
+                ||np.dot(A, x) - b||
+            subject to:
+                x[i] is an element of [0, 1]
+
+        and different algorithms may be substituted in.
+
         Args:
-            target_remove: dictionary returned from Balancer._get_target_remove
-            helper_list: return value of Balancer._to_helper_list
+            A: the occurrence matrix computed in Balancer._get_occurrence_matrix
+            counts: (vector) the original counts for each value
+            target_count: (int) the target value
 
         Returns:
             a list of integer indices to keep
         '''
-        keep_idxs = list(range(len(helper_list)))
-        remove_idxs = []
-        score = self._get_score(target_remove, helper_list, remove_idxs)
-
-        # # ALGO1 - random
-        # import random
-        # random.seed(1)
-        # for _ in range(10000):
-        #     i = random.randrange(len(keep_idxs))
-        #     idx = keep_idxs[i]
-        #     score2 = self._get_score(
-        #         target_remove,
-        #         helper_list,
-        #         remove_idxs + [idx]
-        #     )
-        #     if score2 < score:
-        #         score = score2
-        #         remove_idxs.append(keep_idxs.pop(i))
-
-        # ALGO2 - greedy
-        while len(keep_idxs) > 0:
-            best_i = 0
-            best_score = self._get_score(
-                target_remove,
-                helper_list,
-                remove_idxs + [keep_idxs[best_i]]
-            )
-            for i in range(1, len(keep_idxs)):
-                cur_score = self._get_score(
-                    target_remove,
-                    helper_list,
-                    remove_idxs + [keep_idxs[i]]
-                )
-                if cur_score < best_score:
-                    best_score = cur_score
-                    best_i = i
-
-            if best_score >= score:
-                break
-            score = best_score
-            remove_idxs.append(keep_idxs.pop(best_i))
+        b = counts - target_count
 
-        return keep_idxs
+        # x = self._random(A, b)
```

Comment:
Remove commented out code.  You can add a `todo` to allow the user to specify the algorithm.